### ISSUE TITLE
Provider is using 'fs', which is not available on the client side

### ISSUE THREAD

**Issue by grumpyoldman-io** [project-serum/anchor#728]
When using anchor in a NextJS project it trips because the provider is using the 'fs' package inside the NodeWallet. Ideally this is prevented by either not loading the with the provider (there already is a isBrowser check) or by not using the 'fs' package at all in this file (perhaps by creating a separate local provider file)

**Comment by Arrowana** [comment]
Why would a front-end use a NodeWallet though?
Use a wallet adapter

```
const provider = new Provider(connection, walletAdapter);

const program = new Program(
    PotatoJSON,
    POTATO_PROGRAM_ID,
    provider
  )

**Comment by grumpyoldman-io** [comment]
I agree, this is also how I use it, but the NodeWallet is part of the Provider.ts file and thus loaded by the frontend. In some stricter environments (like Nextjs) this throws an error.

Ideally the local node provider is not part of this file.

**Comment by tomland123** [comment]
https://youtu.be/7p_8549sinc?t=108 this a video of setting up anchor in the client. Here is a sample repo https://github.com/tomland123/react-anchor-tutorial/blob/master/src/MinimalSetup.tsx

**Comment by PirosB3** [comment]
This happened to me too, thanks @tomland123 for providing a workaround
